src/main/java/hudson/plugins/gradle/injection/MavenInjectionEnvironmentContributor.java

```diff
@@ -0,0 +1,105 @@
+package hudson.plugins.gradle.injection;
+
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.model.Computer;
+import hudson.model.EnvironmentContributor;
+import hudson.model.Executor;
+import hudson.model.Node;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static hudson.plugins.gradle.injection.MavenOptsHandler.MAVEN_OPTS;
+
+@Extension
+public class MavenInjectionEnvironmentContributor extends EnvironmentContributor implements MavenInjectionAware {
+
+    private static final MavenOptsHandler MAVEN_OPTS_HANDLER = new MavenOptsHandler(
```

Comment:
I think it can be moved to the `MavenInjectionAware` and shared as well.

Reply:
Yup, overlooked this one; moved